[Bug]: Test `gradient_factory` fails on i686

### What happened?

Test `gradient_factory` fails on i686. See more at https://github.com/void-linux/void-packages/actions/runs/4597991300/jobs/8121416509. I am suspecting a similar issue to what I've reported a while ago, see https://github.com/brndnmtthws/conky/issues/1463, which was fixed by https://github.com/brndnmtthws/conky/pull/1470.

### Version

1.19.0

### Which OS/distro are you seeing the problem on?

Linux (other)

### Conky config

_No response_

### Stack trace

_No response_

### Relevant log output

```Shell
Start 13: gradient_factory should convert to and from rgb and get the initial value
13/16 Test #13: gradient_factory should convert to and from rgb and get the initial value ...***Failed    0.01 sec
Filters: gradient_factory should convert to and from rgb and get the initial value

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test-conky is a Catch v2.13.9 host application.
Run with -? for options

-------------------------------------------------------------------------------
gradient_factory should convert to and from rgb and get the initial value
  rgb_gradient_factory
-------------------------------------------------------------------------------
/builddir/conky-1.19.0/tests/test-gradient.cc:95
...............................................................................

/builddir/conky-1.19.0/tests/test-gradient.cc:101: FAILED:
  REQUIRE( result == colour )
with expansion:
  {?} == {?}

-------------------------------------------------------------------------------
gradient_factory should convert to and from rgb and get the initial value
  hsv_gradient_factory
-------------------------------------------------------------------------------
/builddir/conky-1.19.0/tests/test-gradient.cc:106
...............................................................................

/builddir/conky-1.19.0/tests/test-gradient.cc:112: FAILED:
  REQUIRE( result == colour )
with expansion:
  {?} == {?}

-------------------------------------------------------------------------------
gradient_factory should convert to and from rgb and get the initial value
  hcl_gradient_factory
-------------------------------------------------------------------------------
/builddir/conky-1.19.0/tests/test-gradient.cc:117
...............................................................................

/builddir/conky-1.19.0/tests/test-gradient.cc:123: FAILED:
  REQUIRE( result == colour )
with expansion:
  {?} == {?}

===============================================================================
test cases: 1 | 1 failed
assertions: 3 | 3 failed


      Start 14: scan_graph correctly parses input strings
14/16 Test #14: scan_graph correctly parses input strings ...................................   Passed    0.01 sec
      Start 15: scan_command correctly parses input strings
15/16 Test #15: scan_command correctly parses input strings .................................   Passed    0.01 sec
      Start 16: parse_color correctly parses colours
16/16 Test #16: parse_color correctly parses colours ........................................   Passed    0.01 sec

94% tests passed, 1 tests failed out of 16

Total Test time (real) =   0.11 sec

The following tests FAILED:
	 13 - gradient_factory should convert to and from rgb and get the initial value (Failed)
Errors while running CTest
FAILED: CMakeFiles/test.util 
cd /builddir/conky-1.19.0/build && /usr/bin/ctest --force-new-ctest-process
ninja: build stopped: subcommand failed.
```

--- a/src/gradient.cc
+++ b/src/gradient.cc
@@ -57,6 +57,7 @@ Colour gradient_factory::convert_to_rgb(long *const array) {
   c.red = scaled_rgb[0] / SCALE;
   c.green = scaled_rgb[1] / SCALE;
   c.blue = scaled_rgb[2] / SCALE;
+  c.alpha = 255;
 
   return c;
 }

--- a/tests/test-gradient.cc
+++ b/tests/test-gradient.cc
@@ -32,8 +32,11 @@
 #include <conky.h>
 #include <gradient.h>
 
+#include <iomanip>
+#include <iostream>
+
 const int width = 4;
-const Colour colour = Colour::from_argb32(0x996633);  // brown
+const Colour colour = Colour::from_argb32(0xff996633);  // brown
 const long expected_hue = 256;
 const long expected_value = 0x99;   // max(0x99, 0x66, 0x33)
 const long expected_chroma = 0x66;  // (0x99 - 0x33)
@@ -44,6 +47,15 @@ const long expected_green = 0x66;
 const long expected_blue = 0x33;
 
 const long full_scale = conky::gradient_factory::SCALE360;
+
+std::ostream& operator<<(std::ostream& s, const Colour& c) {
+  s << '#';
+  s << std::setfill('0');
+  s << std::setw(2);
+  s << std::setbase(16);
+  s << (int)c.alpha << (int)c.red << (int)c.green << (int)c.blue;
+  return s;
+}
 
 TEST_CASE("gradient_factory::convert_from_rgb returns correct value") {
 #ifdef BUILD_X11

